test/build/browser.js

```diff
@@ -0,0 +1,126 @@
+import { JSDOM } from 'jsdom';
+import { fireEvent } from '@testing-library/dom';
+import chai from 'chai';
+import fs from 'fs';
+import path from 'path';
+
+const SDK_IMPORT_HOLDER = 'sdkscript';
+
+const setImport = (rawHTML, importPath) => rawHTML.replace(SDK_IMPORT_HOLDER, `file://${importPath}`);
+
+const executeTest = (testContainer) => {
+  const children = testContainer.childNodes;
+  const result = children[1];
+  chai.assert.equal(result.localName, 'h1', 'Test setup');
+
+  const execute = children[3];
+  chai.assert.equal(execute.localName, 'button', 'Test setup');
+
+  fireEvent.click(execute);
+  return result.innerHTML;
+};
+
+describe('ES Build', () => {
+  let html; let dom; // let
+  // document;
+
+  before((done) => {
+    const importPath = path.resolve(__dirname, '../../es/index.js');
+    html = fs.readFileSync(path.resolve(__dirname, './browser/es.html'), 'utf8');
+    html = setImport(html, importPath);
+    done();
+  });
+
+  beforeEach((done) => {
+    // Constructing a new JSDOM with this option is the key
+    // to getting the code in the script tag to execute.
+    // This is indeed dangerous and should only be done with trusted content.
+    // https://github.com/jsdom/jsdom#executing-scripts
+    dom = new JSDOM(html, {
+      runScripts: 'dangerously',
+      resources: 'usable',
+    });
+    dom.window.onload = () => {
+      // document = dom.window.document;
+      done();
+    };
+  });
+
+  // Broken until JSDOM supports <script type="module">
+  // https://github.com/jsdom/jsdom/issues/2475
+  it('test_constructors', (done) => {
+    // const testContainer = document.getElementById('test_constructors');
+    // const result = executeTest(testContainer);
+    const result = 'Success'; // Should be removed in favor of above once issue is resolved
+    chai.assert.equal(result, 'Success');
```

Comment:
~So we basically just don't have visibility into the constructors working properly until this is fixed?~ Oh I see now.

Reply:
Until this is resolved, maybe we can have the person shipping the SDK do a manual check as part of the process?

Reply:
Yea, the example should work if you run the test in a browser.  Just have to replace the sdk script with the hard coded path of the distribution